tests/src/Context/DrupalMailContext.php

```diff
@@ -0,0 +1,159 @@
+   *
+   * @var array
+   */
+  protected $variables = array();
+
+  /**
+   * The captured mail data.
+   *
+   * @var array
+   */
+  protected $capturedMail;
+
+  /**
+   * The variable context.
+   *
+   * @var VariableContext
+   */
+  protected $variableContext;
+
+  /**
+   * Constructs a DrupalMailContext object.
+   *
+   * @param DrupalDriverManager $drupal
+   *   The Drupal driver manager.
+   */
+  public function __construct(DrupalDriverManager $drupal) {
+    $this->drupal = $drupal;
+  }
+
+  /**
+   * Grabs other contexts we rely on from the environment.
+   *
+   * @BeforeScenario @internalMail
+   */
+  public function gatherContexts(BeforeScenarioScope $scope) {
+    $environment = $scope->getEnvironment();
+    $this->variableContext = $environment->getContext(VariableContext::class);
+  }
+
+  /**
+   * Clear the captured data.
+   *
+   * @AfterScenario @internalMail
+   */
+  public function clearCapturedData(AfterScenarioScope $scope) {
+    // Deleting the 'drupal_test_email_collector' variable.
+    variable_del('drupal_test_email_collector');
+  }
+
+  /**
+   * Configure Drupal to use the DrupalMailContext for mail handling.
+   *
+   * @Given platform is configured to use the internal mail handling
+   */
+  public function platformIsConfiguredToUseInternalMailHandling() {
+    $mail_system = array('default-system' => 'TestingMailSystem');
+    // Using VariableContext to set the specific variable. It will be
+    // restored to the previous value after finishing a scenario.
+    $this->variableContext->setVariable('mail_system', $mail_system);
+  }
+
+  /**
+   * Verifies that Drupal internal mail handler received the mail.
+   *
+   * @Then the e-mail has been sent
+   */
+  public function assertInternalMailHandlerReceivedTheMail() {
+    $captured_mail = $this->getCapturedMail();
+    assert($captured_mail, isNotEmpty());
+    $this->capturedMail = $captured_mail;
+  }
+
+  /**
+   * Assert the properties in the captured mail.
+   *
+   * @Then the sent e-mail has the following properties:
+   */
+  public function assertCapturedMailHasTheFollowingProperties(TableNode $table) {
+    $this->assertCapturedMail();
+    $expected_properties = $table->getRowsHash();
+    foreach ($expected_properties as $key => $value) {
+      assert($this->capturedMail[$key], equals($value), "Value of '{$key}' is not equal.");
+    }
+  }
+
+  /**
+   * Assert a mail was captured.
+   *
+   * @throws \Exception
+   *   When no mail was captured.
+   */
+  protected function assertCapturedMail() {
+    if (!$this->capturedMail) {
+      throw new \Exception(
+        'The internal mail handler first needs to confirm that it captured the mail'
+      );
+    }
+  }
+
+  /**
+   * Returns a mail captured by the Drupal internal mail handler.
+   *
+   * @return array
+   *   An empty or a mail data array
+   */
+  private function getCapturedMail() {
+    // During the first run with the clean state the variable_get function
```

Comment:
It is because I'm not setting up `drupal_test_email_collector` variable like in the given example. If you think that we could do that please consider adding the small change to the function platformIsConfiguredToUseInternalMailHandling().